src/main/scala/com/databricks/labs/overwatch/ApiCallV2.scala

```diff
@@ -0,0 +1,549 @@
+package com.databricks.labs.overwatch
+
+import com.databricks.labs.overwatch.utils._
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions._
+import scalaj.http.{Http, HttpOptions, HttpResponse}
+
+import java.util
+
+/**
+ * Companion object for APICallV2.
+ */
+object ApiCallV2 extends SparkSessionWrapper {
+
+  def apply(apiEnv: ApiEnv, apiName: String) = {
+    new ApiCallV2(apiEnv)
+      .setApiName(apiName)
+  }
+
+  def apply(apiEnv: ApiEnv, apiName: String, queryJsonString: String) = {
+    new ApiCallV2(apiEnv)
+      .setApiName(apiName)
+      .setQuery(queryJsonString)
+  }
+
+  def apply(apiEnv: ApiEnv, apiName: String, queryJsonString: String, tempSuccessPath: String) = {
+    new ApiCallV2(apiEnv)
+      .setApiName(apiName)
+      .setQuery(queryJsonString)
+      .setSuccessTempPath(tempSuccessPath)
+  }
+
+}
+
+/**
+ * Api class is used to perform the API calls to different API end points and give the result back in the form of DataFrame.
+ *
+ * @param apiEnv param contains the workspace url and the PAT token which will be used to create the URL and authenticate the request.
+ */
+class ApiCallV2(apiEnv: ApiEnv) extends SparkSessionWrapper {
+
+  import spark.implicits._
+
+  private val logger: Logger = Logger.getLogger(this.getClass)
+  private var _endPoint: String = _ //API end point.
+  private var _jsonQuery: String = _ //Extra parameters for API request.
+  private var _apiResponseArray: util.ArrayList[String] = _ //JsonArray containing the responses from API call.
+  private var _serverBusyCount: Int = 0 // Keep track of 429 error occurrence.
+  private var _token: String = _ //Authentication token for API request.
+  private var _successTempPath: String = _ //Unique String which is used as folder name in a temp location to save the responses.
+  private var _unsafeSSLErrorCount = 0; //Keep track of SSL error occurrence.
+  private var _apiMeta: ApiMeta = null //Metadata for the API call.
+  private var _debugFlag = false //Debug flag to print the information when required. d
+  private var _allowUnsafeSSL = false //Flag to make the unsafe ssl.
+  private var _jsonKey: String = "" //Key name for pagination.
+  private var _jsonValue: String = "" //Key value for pagination.
+  private val readTimeoutMS = 60000 //Read timeout.
+  private val connTimeoutMS = 10000 //Connection timeout.
+  private var _printFlag: Boolean = true
+
+  def endPoint: String = _endPoint
+
+  def jsonQuery: String = _jsonQuery
+
+  def serverBusyCount: Int = _serverBusyCount
+
+  def printFlag: Boolean = _printFlag
```

Comment:
It seems like some of these getters are useless for a customer and are better off not being exposed. We generally want to minimize the autoComplete options for users. Let's remove all the getters that are very deep internals such as printFlag, jsonKey, jsonValue, apiResponseArray etc...these are only used for internal control and shouldn't be directly accessed by the user from what i understand.